src/gallery/gestures.js

```diff
@@ -0,0 +1,208 @@
+const scaleMin = 1
+const scaleMax = 2
+const clientWidth = window.innerWidth
+const clientHeight = window.innerHeight
+const temp = {
+	screenX: null,
+	originalMarginLeft: null,
+	currentMarginLeft: null,
+	originalTransition: null,
+	imgOriginalTransition: null,
+	durationStart: null,
+	translateX: 0,
+	translateY: 0,
+	clientX: null,
+	clientY: null
+}
+let evCache = []
+let prevDiff = -1
+let zoomedIn = false
+
+const grabImageFromEvent = ( e ) => {
+	return e.target.nodeName === 'IMG' ? e.target : e.target.querySelector( 'img' )
+}
+
+const grabScaleFromTransform = ( transform ) => {
+	return Number( transform.slice( transform.indexOf( 'scale' ) + 6, -1 ) )
+}
+
+const isInvalidEvent = ( e, items, current, captionText, prefixClassname ) => {
+	return e.target.className === captionText &&
+			items[ current ].querySelector( `.${prefixClassname}-item-caption-expand-cue` ) ||
+			e.target.className === captionText ||
+			e.pointerType !== 'touch' ||
+			e.target.className.indexOf( 'slider-button' ) > -1
```

Comment:
`e.target.className === captionText` is duplicated in here.

correct me if i'm wrong, it seems like ``items[ current ].querySelector( `.${prefixClassname}-item-caption-expand-cue` )`` doesn't really needed here

maybe we can make this method better like given

```js
// example
const isInvalidevent = ( e, invalidClassNameArray ) => {
    return e.pointerType !== 'touch' || // first detect touch
        'algorithm' // e.target.className or classList contains one from the invalidClassNameArray
}
```

Reply:
Yea good idea! I sort of ported this method over from the previous logic and it was in much need of a refactor. Is it better now?

Reply:
Nice, exactly what I mean here.